data_structures/binary_search_tree.ts

```diff
@@ -107,7 +107,7 @@ export class BinarySearchTree<T> implements Iterable<T> {
   constructor(compare: (a: T, b: T) => number = ascend) {
     if (typeof compare !== "function") {
       throw new TypeError(
-        "compare must be a function, did you mean to call BinarySearchTree.from?",
+        "The 'compare' parameter must be a function, did you mean to call BinarySearchTree.from?",
```

Comment:
Ditto
```suggestion
        "The 'compare' parameter must be a function. Did you mean to call BinarySearchTree.from?",
```

Reply:
I also suggest we should prepend `Cannot construct a BinarySearchTree:`

Reply:
```suggestion
        "Cannot construct a BinarySearchTree: the 'compare' parameter must be a function: did you mean to call BinarySearchTree.from?",
```